scripts/wawCommons.py

```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
             replacedValuesNumber += 1
     return targetJson, replacedValuesNumber
 
+def getFunctionResponseJson(CFNamespaceUrl, username, password, package, functionName, parameters, data):
+    functionCallUrl = CFNamespaceUrl + '/actions/' + package + '/' + functionName + parameters
+    logger.info("Calling function url '%s'", functionCallUrl)
+
+    functionResponse = requests.post(functionCallUrl, auth=(username, password),
+                                 headers={'Content-Type': 'application/json',
+                                          'accept': 'application/json'},
+                                 data=data)
+
+    if functionResponse.status_code == 200:
+
+        responseContentType = functionResponse.headers.get('content-type')
+        if responseContentType != 'application/json':
+            logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+            return None
+
+        return functionResponse.json()
+
+    elif functionResponse.status_code == 202: # try once more
+        # 202 Accepted activation request (should not happen while sending 'blocking=true&result=true')
+        logger.warning("Did not receive response from function '%s' in package '%s', trying once more.", functionName, package)
+        logger.info(json.dumps(functionResponse.json()))
+        responseJson = functionResponse.json()
+        activationId = responseJson['activationId']
+        functionCallUrl = CFNamespaceUrl + '/activations/' + activationId + '/result'
+        logger.info("Trying to get function result from url '%s'", functionCallUrl)
+        functionResponse = requests.get(functionCallUrl, auth=(username, password),
+                                        headers={})
+        if functionResponse.status_code == 200:
+
+            responseContentType = functionResponse.headers.get('content-type')
+            if responseContentType != 'application/json':
+                logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+                return None
+
+            responseJson = functionResponse.json()
+            if not 'result' in responseJson or not responseJson['result'] or not 'payload' in responseJson['result']:
```

Comment:
I'm not sure how often can happen that we get bad response format... but if we want to check all in the IF statement... we should also probably check that responseJson['result'] is dict, just for completeness

Reply:
Added.